pkg/controller/azure/compute/akscluster.go

```diff
@@ -404,39 +410,45 @@ func (r *Reconciler) servicePrincipalSecret(instance *computev1alpha1.AKSCluster
 	return newSPSecretValue.String(), nil
 }
 
-func (r *Reconciler) setConnectionSecret(instance *computev1alpha1.AKSCluster, aksClient *azureclients.AKSSetupClient) error {
-	secretName := instance.ConnectionSecretName()
-	_, err := r.clientset.CoreV1().Secrets(instance.Namespace).Get(secretName, metav1.GetOptions{})
-	if err == nil {
-		// secret is already created, nothing more to do
-		return nil
-	} else if !errors.IsNotFound(err) {
-		return err
+func (r *Reconciler) connectionSecret(instance *computev1alpha1.AKSCluster, client *azureclients.AKSSetupClient) (*v1.Secret, error) {
+	creds, err := client.ListClusterAdminCredentials(ctx, *instance)
+	if err != nil {
+		return nil, err
 	}
 
-	// the secret doesn't exist yet, create it now. First look up the cluster's admin credentials
-	clusterCreds, err := aksClient.AKSClusterAPI.ListClusterAdminCredentials(ctx, *instance)
+	// TODO(negz): It's not clear in what case this would contain more than one kubeconfig file.
+	// https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/listclusteradmincredentials#credentialresults
+	if creds.Kubeconfigs == nil || len(*creds.Kubeconfigs) == 0 || (*creds.Kubeconfigs)[0].Value == nil {
+		return nil, fmt.Errorf("zero kubeconfig credentials returned")
+	}
+	yml, err := base64.StdEncoding.DecodeString(string(*(*creds.Kubeconfigs)[0].Value))
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("cannot decode kubeconfig file: %s", err)
```

Comment:
i've consistently used `%+v` for wrapping errors, what do you think about that instead of `%s`?

Reply:
I also prefer `%+v` but mistakenly thought the established pattern in this file was to use `%s`. Fixed.